src/androidTest/java/de/dennisguse/opentracks/settings/PreferencesOpenHelperTest.java

```diff
@@ -0,0 +1,54 @@
+package de.dennisguse.opentracks.settings;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import android.content.Context;
+import android.content.SharedPreferences;
+
+import androidx.preference.PreferenceManager;
+import androidx.test.core.app.ApplicationProvider;
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.content.data.Layout;
+import de.dennisguse.opentracks.util.CsvConstants;
+
+@RunWith(AndroidJUnit4.class)
+public class PreferencesOpenHelperTest {
+
+    private final Context context = ApplicationProvider.getApplicationContext();
+
+    @Test
+    public void test_upgradeFrom1To2() {
+        // given the version 1
+        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+
+        String oldCustomLayoutCsv = context.getString(R.string.activity_type_unknown) + CsvConstants.ITEM_SEPARATOR
+                + "distance,1,1,0" + CsvConstants.ITEM_SEPARATOR + "speed,1,1,0" + CsvConstants.ITEM_SEPARATOR;
```

Comment:
In the tests, ITEM_SEPARATOR must not be referenced.
It is easier to read if not :)

Reply:
Okay, I did it, but I don't understand why must not reference to ITEM_SEPARATOR :thinking: because if we change it then we have to adjust all tests.

Reply:
More a philosophical thing: tests should not rely on implementation details (in this case CsvConstants).
We do this in some place, but it is nicer to not do it, i.e., if somebody changes ITEM_SEPARATOR the tests should fail (makes the tests also easier to read).

There is a tradeoff writing tests and not writing them, as they require maintenance :man_shrugging:

Reply:
Okay, I understand. Thank you so much for the clarification ;)